core/amber/src/main/scala/edu/uci/ics/texera/web/resource/dashboard/workflow/WorkflowExecutionsResource.scala

```diff
@@ -71,4 +81,32 @@ class WorkflowExecutionsResource {
         .toList
     }
   }
+
+  /** Sets a single execution's bookmark to the payload passed in the body. */
+  @PUT
+  @Path("/set_execution_bookmark")
+  @Consumes(Array(MediaType.APPLICATION_JSON))
+  def setExecutionIsBookmarked(
+      request: ExecutionBookmarkRequest,
+      @Auth sessionUser: SessionUser
+  ): Unit = {
+    validateUserCanAccessWorkflow(request.wid, sessionUser.getUser.getUid)
+    getExecutionById(request.eId) match {
```

Comment:
better not make it optional since the eid is coming from the list itself not user entered.

Reply:
See above